lib/rubocop/cop/lint/redundant_with_index.rb

```diff
@@ -56,10 +56,10 @@ def on_block(node)
         def_node_matcher :redundant_with_index?, <<~PATTERN
           {
             (block
-              $(call _ {:each_with_index :with_index} ...)
+              $(call ^_ {:each_with_index :with_index} ...)
               (args (arg _)) ...)
             (numblock
-              $(call _ {:each_with_index :with_index} ...) 1 ...)
+              $(call ^_ {:each_with_index :with_index} ...) 1 ...)
```

Comment:
I think it might be clearer to write `return unless node.receiver` at the beginning of `def on_block` rather than using `^`.
Ref: https://github.com/rubocop/rubocop/pull/12768#discussion_r1518852124